custom/aaa/views.py

```diff
@@ -177,14 +178,22 @@ def post(self, request, *args, **kwargs):
                 dict(name='test 6', age='19', gender='M', lastImmunizationType=1, lastImmunizationDate='2018-03-03'),
             ]
         elif beneficiary_type == 'eligible_couple':
-            data = [
-                dict(name='test 1', age='17', currentFamilyPlanningMethod=1, adoptionDateOfFamilyPlaning='2018-03-03'),
-                dict(name='test 2', age='57', currentFamilyPlanningMethod=0, adoptionDateOfFamilyPlaning='2018-03-03'),
-                dict(name='test 3', age='25', currentFamilyPlanningMethod=0, adoptionDateOfFamilyPlaning='2018-03-03'),
-                dict(name='test 4', age='16', currentFamilyPlanningMethod=1, adoptionDateOfFamilyPlaning='2018-03-03'),
-                dict(name='test 5', age='22', currentFamilyPlanningMethod=0, adoptionDateOfFamilyPlaning='2018-03-03'),
-                dict(name='test 6', age='31', currentFamilyPlanningMethod=1, adoptionDateOfFamilyPlaning='2018-03-03'),
-            ]
+            data = (
+                Woman.objects
+                .annotate(
+                    age=ExtractYear(Func(F('dob'), function='age')),
+                )
+                .filter(
+                    # should filter for date
+                    # should filter for location
+                    age__range=(19, 49),
+                    marital_status='married',
+                )
+                .exclude(migration_status='yes')
+                .extra(select={'currentFamilyPlanningMethod': 0, 'adoptionDateOfFamilyPlaning': '2018-03-01'})
+                .values('person_case_id', 'name', 'age', 'currentFamilyPlanningMethod', 'adoptionDateOfFamilyPlaning')
+            )[:10]
+            data = list(data)
```

Comment:
@lwyszomi I halfway got this to work. I have left out some of the filtering. Can you pick this up as you finish this view?